atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/precommit/DefaultPreCommitRequirementValidator.java

```diff
@@ -70,15 +69,14 @@ public void throwIfPreCommitConditionInvalidAtCommitOnWriteTransaction(
     }
 
     @Override
-    public void throwIfPreCommitRequirementsNotMet(LockToken commitLocksToken, long timestamp) {
-        throwIfImmutableTsOrCommitLocksExpired(commitLocksToken);
+    public void throwIfPreCommitRequirementsNotMet(long timestamp) {
```

Comment:
Called from:
* SnapshotTransaction#abort; already reasoned through this call.
* DefaultReadSnapshotValidator#throwIfPreCommitRequirementsNotMetOnRead: this is called from a bunch of places on the read path; e.g. SnapshotTransaction#validatePreCommitRequirementsOnReadIfNecessary, DefaultKeyValueSnapshotReader#collectCellsToPostFilter - These are equivalent calls, whilst we haven't finished a previous refactoring. In any case, there should be no extra locks in the manager at the point of this call, but even if there are somehow (because we migrated AdvisoryLocks to this) we can see that Advisory locks are also checked on this codepath, so it's all good.
* SnapshotTransaction#throwIfValueChangedConflict: already reasoned through this call.

Reply:
In general, for now at least for calls with _null_ this should be semantics preserving. Going further, it would mean that we might check user specified locks on these reads, but I think that's fine.